ui/test/nuxeo-aggregation-widget.test.js

```diff
@@ -55,42 +131,235 @@ const data = {
+      expect(isElementVisible(icon)).to.be.true;
+    });
+
+    test('Should hide checkboxes when "collapsible" is set', async () => {
+      const nuxeoCheckboxAggregation = await fixture(
+        html`
+          <nuxeo-checkbox-aggregation .data=${data} collapsible></nuxeo-checkbox-aggregation>
+        `,
+      );
+
+      const checkboxes = nuxeoCheckboxAggregation.shadowRoot.querySelectorAll('paper-checkbox');
+      checkboxes.forEach((element) => {
+        expect(isElementVisible(element)).to.be.false;
+      });
+    });
+
+    test('Should display 8 checkboxes when "collapsible" and "opened" are set', async () => {
+      const nuxeoCheckboxAggregation = await fixture(
+        html`
+          <nuxeo-checkbox-aggregation .data=${data} collapsible opened></nuxeo-checkbox-aggregation>
+        `,
+      );
+
+      const checkboxes = nuxeoCheckboxAggregation.shadowRoot.querySelectorAll('paper-checkbox');
+      checkboxes.forEach((element) => {
+        expect(isElementVisible(element)).to.be.true;
+      });
+    });
+
+    test('Should display checkboxes when "collapsible" is set and it is tapped', async () => {
+      const nuxeoCheckboxAggregation = await fixture(
+        html`
+          <nuxeo-checkbox-aggregation .data=${data} collapsible></nuxeo-checkbox-aggregation>
+        `,
+      );
+
+      const checkboxes = nuxeoCheckboxAggregation.shadowRoot.querySelectorAll('paper-checkbox');
+      expect(checkboxes.length).to.equal(8);
+      checkElementsVisibility(checkboxes, 8, false);
+
+      tap(nuxeoCheckboxAggregation.shadowRoot.querySelector('button'));
+      expect(checkboxes.length).to.equal(8);
+      checkElementsVisibility(checkboxes, 8, true);
+    });
+
+    test('Should display 4 checkboxes when "visibleItems" is set to 4', async () => {
+      const nuxeoCheckboxAggregation = await fixture(
+        html`
+          <nuxeo-checkbox-aggregation .data=${data} collapsible visible-items="4"></nuxeo-checkbox-aggregation>
+        `,
+      );
+
+      const checkboxes = nuxeoCheckboxAggregation.shadowRoot.querySelectorAll('paper-checkbox');
+      expect(checkboxes.length).to.equal(4);
+      checkElementsVisibility(checkboxes, 4, false);
+
+      tap(nuxeoCheckboxAggregation.shadowRoot.querySelector('button'));
+      checkElementsVisibility(checkboxes, 4, true);
+    });
+
+    test('Should show 9 checkboxes when "show all" button is tapped', async () => {
+      const nuxeoCheckboxAggregation = await fixture(
+        html`
+          <nuxeo-checkbox-aggregation .data=${data} collapsible label="asdklfj"></nuxeo-checkbox-aggregation>
+        `,
+      );
+
+      let checkboxes = nuxeoCheckboxAggregation.shadowRoot.querySelectorAll('paper-checkbox');
+      expect(checkboxes.length).to.equal(8);
+      checkElementsVisibility(checkboxes, 8, false);
+
+      tap(nuxeoCheckboxAggregation.shadowRoot.querySelector('button'));
+      // for iron-collapse animation to end
+      await timePasses(300);
```

Comment:
I see that we have this after this tap, but we don't on the previous couple of tests. Could this possibly add some flakiness to the tests? We could introduce a method to do the tapping and the waiting and re-use in on the tests, to encapsulate this logic and make sure we always take advance of it.